### DISCUSSION TITLE
NEXT_DATA html file size.

### DISCUSSION THREAD

**Discussion by lucashfreitas** [vercel/next.js#33926] [Help]
Hello, I understand that NextJS needs to use NEXT_DATA to hydrate the React components in the client side as stated here: https://github.com/vercel/next.js/discussions/15117 , https://github.com/vercel/next.js/discussions/13418#, https://www.youtube.com/watch?v=kJNDNYXYvPk. 

Is there an option to load those files on demand using FETCH as `gatsby` does? We could add a prefetch `<link>` and fetch those json files from the server on demand without adding it on HTML straightaway? I am aware that it might slow down the rendering a little bit as we are adding another HTTP request but on the other side, the RESPONSE SIZE of the HTML file would be smaller compared to pages that have massive JSON objects as the data. (I know we can try to clean, reduce the page data object but some huge websites the page does needs a lot of data, specially for markdown, so naturally the HTML size increases a lot due duplication of the data in `NEXT_DATA`).

Thanks,

**Comment by Bahlaouane-Hamza** [comment]
Hi,
You will need to patch Next.js for this, here's one for next@12.1.6
You can use https://www.npmjs.com/package/patch-package to auto install the patch.

[next+12.1.6.patch](https://github.com/vercel/next.js/files/11009343/next%2B12.1.6.patch)

For this part
```js
+    delete props1.pageProps.node
+    delete props1.pageProps.i18n
+    delete props1.pageProps.menus
```
I'm removing only the props i don't wanna keep in the HTML, If you are using NextAuth for example, you want to keep `session` coming from SSR, and this may vary depending on your use case.

And finally on the _hydrate part, i just restore the props i removed.

Hope this helps.
